src/core.js

```diff
@@ -2846,6 +2870,10 @@ Strophe.Connection = class Connection {
         }
         Strophe.info("SASL authentication succeeded.");
 
+        if (this._sasl_data.keys) {
+            this.scramKeys = this._sasl_data.keys;
```

Comment:
It's not clear to me why you first store the keys on `this._sasl_data.keys` just to then set them again on `this.scramKeys`.

Why not just set them on `this.scramKeys` from the start?

Reply:
`scramKeys` is supposed to be the public interface to the keys. For instance, I have a fork of converse which uses this for automatic login. My intent here was that this variable would be non null if and only if the authentication succeeded, while `this._sasl_data.keys` would be used internally. As it is now, `this._sasl_data.keys` will be non null if the server rejects the supplied proof. I'm open to changes here, or at the very least adding a comment explaining this.

Reply:
Ok, makes sense.